privacyidea/lib/token.py

```diff
@@ -892,8 +893,8 @@ def _gen_serial(_prefix, _tokennum):
         num_str = '{:04d}'.format(_tokennum)
         h_len = serial_len - len(num_str)
         if h_len > 0:
-            h_serial = binascii.hexlify(os.urandom(h_len)).upper()[0:h_len]
-        return "{0!s}{1!s}{2!s}".format(_prefix, num_str, h_serial)
+            h_serial = hexlify_and_unicode(os.urandom(h_len)).upper()[0:h_len]
+        return u"{0!s}{1!s}{2!s}".format(_prefix, num_str, h_serial)
```

Comment:
The serial number will never contain a non-ascii character.
So should we omit the ``u""`` since Python 3 also would not recommend it?